iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -34,12 +35,33 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
             return new DigitalTwinClient(iotHubConnectionString.HttpsEndpoint, sharedAccessKeyCredential, handlers);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DigitalTwinClient"/> class.</summary>
+        /// <param name="connectionString">The IoT hub's connection string.</param>
+        /// <param name="transportSettings">The optional settings to apply to the underlying http client.</param>
+        public static DigitalTwinClient CreateFromConnectionString(string connectionString, HttpTransportSettings transportSettings)
```

Comment:
Strangely, we didn't have a constructor for our digital twin client that took in the HttpTransportSettings options, so I created one

Reply:
I wonder if this will be confusing, since the recommended approach to enable proxy support (a part of HttpTransportSetting) is to do so via a delegating handler; and now, even though we take in an HttpTransportSetting, we can't really set proxy that way.

How does track 2 handle this? They also have concepts of pipeline for the http client, so how do they accept other custom values that are set on the client directly and not via the pipeline.

Reply:
Also, if we decide to go this way we will need a constructor that takes in both the transport settings and the delegating handlers.

Reply:
> How does track 2 handle this? They also have concepts of pipeline for the http client, so how do they accept other custom values that are set on the client directly and not via the pipeline.

IIRC they make each library extend the pipelineoptions class to include whatever library specific options there are. In this case, we already don't have that approach and it seems a bit odd to add it now.

The way I see it, the delegating handlers constructor is for users who want complete control over all aspects of the http pipeline that our pnp client will use. This new constructor falls more in line with the other service client constructors we have.

Reply:
To expand on this a bit, the only constructors in our PL that accept this bug fix are the ones that take in the httpClient rather than taking in any delegating handlers. 

```csharp
// constructors like these allow us to make the httpClient use the SocketsHttpHandler for .netcore cases
protected IotHubGatewayServiceAPIs(HttpClient httpClient, bool disposeHttpClient)

// constructors like these don't accept SocketsHttpHandler since it doesn't extend DelegatingHandler and doesn't extend HttpClientHandler
protected IotHubGatewayServiceAPIs(HttpClientHandler rootHandler, params DelegatingHandler[] handlers)
```

With that in mind, we are forced to own the httpClient in order to fix this issue

Reply:
> Also, if we decide to go this way we will need a constructor that takes in both the transport settings and the delegating handlers.

This will just lead to confusion since there would be 2 ways to set the proxy, and we don't want that.

Reply:
> The other reason being that there isn't a way for me to build an httpclient that uses all the delegating handlers.

Isn't this the approach that you had in the other PR? 

Also, even if we do decide to separate the 2 approaches (full control over http client vs we'll set up most of it for you), we need to explicitly specify that even though we have a constructor that takes in HttpTransportSettings, and thus proxy settings, you cannot set proxy up that way.
We will also need to provide some input (doc/ redme/ blog post) on how someone can set up both proxy (and logging, or any other functionality achieved via delegating handler) and also get dns refresh support.

Reply:
> we need to explicitly specify that even though we have a constructor that takes in HttpTransportSettings, and thus proxy settings, you cannot set proxy up that way.

This PR does include setting the proxy to the local http client if one is specified in the HttpTransportSettings

Reply:
> > The other reason being that there isn't a way for me to build an httpclient that uses all the delegating handlers.

> Isn't this the approach that you had in the other PR?

Yes, but I was never fully comfortable with the chaining of delegating handlers that I originally had. As you rightly pointed out, each delegating handler may have some number of inner handlers already. Trying to code around that is a hassle, and I don't see much reason to do so.

Reply:
> This PR does include setting the proxy to the local http client if one is specified in the HttpTransportSettings

It does? Then mark me satisfied!